### PR TITLE
nanodbc

### PR DESCRIPTION
**Pull request by derekjsonar**

### CODE REVIEW

File: SonarExternal.cmake [hunk 1]
```diff
@@ -2026,6 +2026,77 @@ function(build_simdjson)
...
+  endif()
+  message(STATUS "Building unixodbc ${UNIXODBC_VERSION}")
+  ExternalProject_Add(unixodbc
+          URL http://www.unixodbc.org/unixODBC-${UNIXODBC_VERSION}.tar.gz
+          URL_MD5 274a711b0c77394e052db6493840c6f9
+          DOWNLOAD_NO_PROGRESS 1
+          BUILD_IN_SOURCE 1
+          CC=${CMAKE_C_COMPILER_LAUNCHER}\ ${CMAKE_C_COMPILER}
+          CXX=${CMAKE_CXX_COMPILER_LAUNCHER}\ ${CMAKE_CXX_COMPILER}
+          CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix <INSTALL_DIR>
```
Comment (yhagerjs) [inline SonarExternal.cmake]: I don't think these `CC` and `CXX` actually do what you think. They need to be part of the `CONFIGURE_COMMAND`

File: SonarExternal.cmake [hunk 2]
```diff
@@ -2026,6 +2026,77 @@ function(build_simdjson)
...
+             <INSTALL_DIR>/lib/libodbccr.so
+          )
+  external_project_dirs(unixodbc install_dir)
+  foreach(obj odbcinst odbc odbccr)
+    add_library(unixodbc::${obj} SHARED IMPORTED GLOBAL)
+    add_dependencies(unixodbc::${obj} unixodbc)
+    set_target_properties(unixodbc::${obj} PROPERTIES IMPORTED_LOCATION ${unixodbc_install_dir}/lib/lib${obj}.so)
+    include_external_directories(TARGET unixodbc::${obj} DIRECTORIES ${unixodbc_install_dir}/include)
+  endforeach()
+endfunction()
```
Comment (yhagerjs) [inline SonarExternal.cmake]: I think we must add `install(FILES` command here, to install the dynamic libraries onto the package.

File: SonarExternal.cmake (hunk 1 above)
Comment (derekjsonar) [reply SonarExternal.cmake]: this is done now

File: SonarExternal.cmake (hunk 2 above)
Comment (derekjsonar) [reply SonarExternal.cmake]: I decided to do that in `gateway/src/CMakeLists.txt`, see the parallel PR

```
install(TARGETS sonargateway RUNTIME DESTINATION lib/sonar/gateway)
## here, this works fine
install(DIRECTORY $<TARGET_FILE_DIR:unixodbc::odbcinst>/
        DESTINATION lib/sonar/gateway)
```
Then I have
```
derek@derek ~/work/builds/gateway/dev6/RelWithDebInfo$ rpm -qpl /home/derek/work/builds/gateway/dev6/RelWithDebInfo/sonargateway-4.1.10-363-g2781e4d.el7.jsonar.x86_64.rpm | grep odbc                                                                                                                                       
/usr/lib/sonar/gateway/libodbc.la
/usr/lib/sonar/gateway/libodbc.so
/usr/lib/sonar/gateway/libodbc.so.2
/usr/lib/sonar/gateway/libodbc.so.2.0.0
/usr/lib/sonar/gateway/libodbccr.la
/usr/lib/sonar/gateway/libodbccr.so
/usr/lib/sonar/gateway/libodbccr.so.2
/usr/lib/sonar/gateway/libodbccr.so.2.0.0
/usr/lib/sonar/gateway/libodbcinst.la
/usr/lib/sonar/gateway/libodbcinst.so
/usr/lib/sonar/gateway/libodbcinst.so.2
/usr/lib/sonar/gateway/libodbcinst.so.2.0.0
/usr/lib/sonar/gateway/pkgconfig/odbc.pc
/usr/lib/sonar/gateway/pkgconfig/odbccr.pc
/usr/lib/sonar/gateway/pkgconfig/odbcinst.pc
```
And finally
```
derek@derek ~/work/builds/gateway/dev6/RelWithDebInfo$ /usr/lib/sonar/gateway/sonargateway -v
sonargateway 4.1.10-364-g6a65142
build-tag: v4.1.10-364-g6a65142
build-type: RelWithDebInfo
revision: 6a651429c3f9fe6b3403d77e4f9d4c5d07ce6409

Given Command Line Options:
  --data-dir: /var/lib/sonar/gateway [default]
  --date-formats: /etc/sonar/gateway/dt_fmt_strings [default]
  --default-log-file: /var/log/sonar/gateway/sonargateway.log [default]
  --delay-messages-only: false [default]
  --flush-interval:  [default]
  --input-delimiter: D3L1M1T3R [default]
  --input-module-state-file:  [default]
  --input-stats-interval:  [default]
  --logging-configuration: /etc/sonar/gateway/gateway-logging.conf [default]
  --output-delimiter:  [default]
  --output-stats-interval:  [default]
  --prenorm-delimiter: PR3N0RM [default]
  --print-stats-on-exit: false [default]
  --redis-host: 127.0.0.1 [default]
  --redis-port: 6379 [default]
  --shutdown-after-x-messages: 0 [default]
  --validate: false [default]
  --version: 

derek@derek ~/work/builds/gateway/dev6/RelWithDebInfo$ ldd /usr/lib/sonar/gateway/sonargateway 
        linux-vdso.so.1 =>  (0x00007ffee9534000)
        libodbcinst.so.2 => /usr/lib/sonar/gateway/libodbcinst.so.2 (0x00007f8538cc7000)
        libodbc.so.2 => /usr/lib/sonar/gateway/libodbc.so.2 (0x00007f8538a54000)
        libodbccr.so.2 => /usr/lib/sonar/gateway/libodbccr.so.2 (0x00007f8538849000)
        libarchive.so.13 => /lib64/libarchive.so.13 (0x00007f85385a2000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f8538389000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f8538181000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f8537f65000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f8537d61000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f8537a5a000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f8537758000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f8537542000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f8537174000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8538ee2000)
        libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007f8536d11000)
        libacl.so.1 => /lib64/libacl.so.1 (0x00007f8536b08000)
        libattr.so.1 => /lib64/libattr.so.1 (0x00007f8536903000)
        liblzo2.so.2 => /lib64/liblzo2.so.2 (0x00007f85366e2000)
        liblzma.so.5 => /lib64/liblzma.so.5 (0x00007f85364bc000)
        libbz2.so.1 => /lib64/libbz2.so.1 (0x00007f85362ac000)
        libxml2.so.2 => /lib64/libxml2.so.2 (0x00007f8535f42000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f8535d2c000)
```
Note the ldd paths for the `odbc` stuff, looks good.

**Pull request merged**